test/framework/shootframework.go

```diff
@@ -0,0 +1,290 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/retry"
+	"github.com/onsi/ginkgo"
+	"github.com/pkg/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	apiextensionsscheme "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/scheme"
+	corescheme "k8s.io/client-go/kubernetes/scheme"
+	apiregistrationscheme "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset/scheme"
+	metricsscheme "k8s.io/metrics/pkg/client/clientset/versioned/scheme"
+)
+
+var shootCfg *ShootConfig
+
+// ShootConfig is the configuration for a shoot framework
+type ShootConfig struct {
+	GardenerConfig *GardenerConfig
+	ShootName      string
+
+	CreateTestNamespace         bool
+	DisableTestNamespaceCleanup bool
+}
+
+// ShootFramework represents the shoot test framework that includes
+// test functions that can be executed ona specific shoot
+type ShootFramework struct {
+	*GardenerFramework
+	TestDescription
+	Config *ShootConfig
+
+	SeedClient  kubernetes.Interface
+	ShootClient kubernetes.Interface
+
+	Seed         *gardencorev1beta1.Seed
+	CloudProfile *gardencorev1beta1.CloudProfile
+	Shoot        *gardencorev1beta1.Shoot
+	Project      *gardencorev1beta1.Project
+
+	Namespace string
+}
+
+// NewShootFramework creates a new simple Shoot framework
+func NewShootFramework(cfg *ShootConfig) *ShootFramework {
```

Comment:
When using the ShootFramework in an extension provider, the seed scheme is different from the Gardener Seed scheme. E.g the mcm api group (Machines, MachineDeployments, ..) does not exist in the Gardener.
We need to have the ability to customize the scheme that is being used to create the seed client. 
Maybe here we can hand over the seed scheme in case we want to override it.

In the extensions we then would create a scheme like this (adding the provider specific groups)

`extensionSeedScheme := kubernetes.SeedScheme
	seedSchemeBuilder := runtime.NewSchemeBuilder(
		machinev1alpha1.AddToScheme,
	)
	utilruntime.Must(seedSchemeBuilder.AddToScheme(extensionSeedScheme))`

Reply:
the seedScheme is now included as shootframwork configuration and will be passed to the `GetSeed` if no scheme is given it is defaulted to `kubernetes.SeedScheme`